node/pegnet/transaction.go

```diff
@@ -0,0 +1,60 @@
+package pegnet
+
+import (
+	"encoding/hex"
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+const (
+	DefaultPad = 3
+)
+
+// SplitTxID splits a TxID into it's parts.
+// TxID format : [TxIndex]-[EntryHash]
+//				 1-c99dedea0e4e0c40118fe7e4d515b23cc0489269c8cef187b4f15a4ccbd880be
+func SplitTxID(txid string) (index int, entryhash string, err error) {
+	arr := strings.Split(txid, "-")
+	if len(arr) != 2 {
+		return -1, "", fmt.Errorf("txid does not match txid format, format: [TxIndex]-[EntryHash]")
+	}
+
+	txIndex, err := strconv.ParseInt(arr[0], 10, 32)
+	if err != nil {
+		return -1, "", fmt.Errorf("index must be a valid integer")
+	}
+	if txIndex < 0 {
+		return -1, "", fmt.Errorf("index must be a postive integer")
+	}
+
+	// Verify the entryhash is valid
+	hash, err := hex.DecodeString(arr[1])
+	if err != nil {
+		return -1, "", fmt.Errorf("entryhash must be a valid hex string")
+	}
+	if len(hash) != 32 {
+		return -1, "", fmt.Errorf("entryhash must be 32 bytes (64 hex characters)")
+	}
+
+	// Over 10k index? That's impossible, that means each tx is of 1 byte
+	if txIndex > 10000 {
```

Comment:
not sure why this check would be in this function? imo this function is for checking syntax but not semantics